### PR TITLE
Add capability to sort objects in instances view by retained size

### PR DESCRIPTION
**Pull request by nikita-nazarov**
This pull request adds a feature that utilises the memory agent to calculate retained sizes of all objects of a specific class. This objects are then displayed in instances view and sorted by their retained size, which allows a user to inspect the heaviest object of a specific class.

@zuevmaxim 

https://github.com/JetBrains/intellij-community/assets/25721619/8c507db8-e157-4f9a-9536-60aab839d2c8

### CODE REVIEW

File: java/debugger/impl/src/com/intellij/debugger/memory/agent/ui/MemoryAgentDialog.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        }
+
+        @Override
+        public void sessionResumed() {
+          close(DialogWrapper.OK_EXIT_CODE);
+        }
+      }, myDisposable);
+    }
+
+    if (session != null) {
```
Comment (zuevmaxim) [inline java/debugger/impl/src/com/intellij/debugger/memory/agent/ui/MemoryAgentDialog.java:78]: It looks like a duplicate by an accident condition, I'll remove it

File: java/debugger/impl/src/com/intellij/debugger/memory/ui/InstancesView.java [hunk 2]
```diff
@@ -388,4 +415,68 @@ public JavaReferenceInfo get(int index) {
...
+
+
+  private final class TreeTableRepresentation implements InstancesViewRepresentation {
+    @Override
+    public void customizeView(@NotNull InstancesView view, @NotNull XDebugSession session, String className) {
+      InstancesViewTreeTableModel treeTableModel = new InstancesViewTreeTableModel(myInstancesTree);
+      JBTreeTable treeTable = new JBTreeTable(treeTableModel, myInstancesTree);
+      treeTable.setDefaultRenderer(Long.class, treeTableModel.createTableCellRenderer());
+      treeTable.getTree().setCellRenderer(new XDebuggerTreeRenderer(myInstancesTree.getProject()));
+      treeTable.setModel(new InstancesViewTreeTableModel(myInstancesTree));
```
Comment (zuevmaxim) [inline java/debugger/impl/src/com/intellij/debugger/memory/ui/InstancesView.java:471]: @nikita-nazarov Why do we need a model reset here?

File: java/debugger/impl/src/com/intellij/debugger/memory/ui/InstancesView.java [hunk 3]
```diff
@@ -388,4 +415,68 @@ public JavaReferenceInfo get(int index) {
...
+      return instances;
+    }
+  }
+
+
+  private final class TreeTableRepresentation implements InstancesViewRepresentation {
+    @Override
+    public void customizeView(@NotNull InstancesView view, @NotNull XDebugSession session, String className) {
+      InstancesViewTreeTableModel treeTableModel = new InstancesViewTreeTableModel(myInstancesTree);
+      JBTreeTable treeTable = new JBTreeTable(treeTableModel, myInstancesTree);
```
Comment (zuevmaxim) [inline java/debugger/impl/src/com/intellij/debugger/memory/ui/InstancesView.java:468]: Is it critical to pass `myInstancesTree ` as a tree parameter to `JBTreeTable`? As I can see, `JBTreeTable` relies on the model, so delegating via `InstancesViewTreeTableModel` looks enough

**Comment by nikita-nazarov** [comment]
Thanks for your comments! Fixed all of them.

**Review by zuevmaxim — APPROVED** [review]

**Pull request closed without merging**